### ISSUE TITLE
Evaluate using Travis for Windows builds

### ISSUE THREAD

**Issue by lassik** [Unibeautify/unibeautify#204]
From the Travis CI blog: [Windows is Available (Early Release)](https://blog.travis-ci.com/2018-10-11-windows-early-release) Oct 11, 2018

Found by @stevenzeck

**Comment by stevenzeck** [comment]
To Do:

- [ ] Add `windows` to the `os` [list](https://github.com/Unibeautify/unibeautify/blob/b622acc09d220c6a50112d41315a0f6209376a96/.travis.yml#L8-L10)
- [ ] Don't download/configure [CodeClimate](https://github.com/Unibeautify/unibeautify/blob/b622acc09d220c6a50112d41315a0f6209376a96/.travis.yml#L16-L18) or send [test results](https://github.com/Unibeautify/unibeautify/blob/b622acc09d220c6a50112d41315a0f6209376a96/.travis.yml#L28) when running Windows builds as it is not supported
- [ ] Configure so all OS's use `LF` and not `CRLF`.  When cloning a repo and checking out a commit, the line endings depend on the OS, so Windows builds will fail TSLint since it uses CRLF and not LF. I'm playing around by adding a `.gitattributes` file with `* text=auto eol=lf`.

**Comment by stevenzeck** [comment]
Seems to be working in beautifier-eslint:

* [.travis.yml](https://github.com/Unibeautify/beautifier-eslint/blob/7d04401f10e4a8763474a95b3e40d5915065d95d/.travis.yml)
* [.gitattributes](https://github.com/Unibeautify/beautifier-eslint/blob/7d04401f10e4a8763474a95b3e40d5915065d95d/.gitattributes)

**Comment by lassik** [comment]
Awesome 🎉 The eslint Windows builds seem to be red, but only because of the CRLF situation.

**Comment by lassik** [comment]
What's different between them? The commit hash is the same.

**Comment by stevenzeck** [comment]
I'll try pushing everything as one commit and see if that makes a difference.  Otherwise I think one is a standard commit while the other is a merge commit.

**Comment by lassik** [comment]
The outcome seems to be the same :/ I just had a weird incident too, where something worked in Travis in a branch but not in a PR: https://github.com/Unibeautify/beautifier-clang-format/pull/81

**Comment by lassik** [comment]
BTW, should Code Climate be run separately for each platform, or is it enough to run it on Linux only?

**Comment by stevenzeck** [comment]
It shouldn't matter.  All it does it reports the coverage results of `npm test` (I think all repos use that command) from Travis to CodeClimate, not much overhead.

**Comment by lassik** [comment]
Ah, that makes sense. For reference, Code Climate is not available (pre-built) for Windows, only Linux and OSX: https://docs.codeclimate.com/v1.0/docs/configuring-test-coverage#section-locations-of-pre-built-binaries

**Comment by stevenzeck** [comment]
Yep, that's the second checkmark https://github.com/Unibeautify/unibeautify/issues/204#issuecomment-429361164

**Comment by lassik** [comment]
Sorry, missed that one.

**Comment by Glavin001** [comment]
👍I would greatly prefer Travis CI to [AppVeyor](https://www.appveyor.com/).

Let me know if you need anything from me. Sounds like you already have a plan and are making this a reality 🎉 ! Thanks!
